pkg/controller/registrysyncer/registrysyncer_test.go

```diff
@@ -0,0 +1,178 @@
+	}{
+		{
+			name:               "app.ci with svc dns",
+			ctx:                "app.ci",
+			potentiallyPrivate: "image-registry.openshift-image-registry.svc:5000/ci/applyconfig@sha256:bf08a76268b29f056cfab7a105c8473b359d1154fbbe3091fe6052ad6d0427cd",
+			expected:           "registry.ci.openshift.org/ci/applyconfig@sha256:bf08a76268b29f056cfab7a105c8473b359d1154fbbe3091fe6052ad6d0427cd",
+		},
+		{
+			name:               "api.ci with svc dns",
+			ctx:                "api.ci",
+			potentiallyPrivate: "docker-registry.default.svc:5000/ci/applyconfig@sha256:bf08a76268b29f056cfab7a105c8473b359d1154fbbe3091fe6052ad6d0427cd",
+			expected:           "registry.svc.ci.openshift.org/ci/applyconfig@sha256:bf08a76268b29f056cfab7a105c8473b359d1154fbbe3091fe6052ad6d0427cd",
+		},
+		{
+			name:               "api.ci with public domain",
+			ctx:                "api.ci",
+			potentiallyPrivate: "gcr.io/k8s-prow/tide@sha256:5245b7747c44d560aab27bc07dbaaf50bbb55f71d0973f85b09c79b8d8b93c97",
+			expected:           "gcr.io/k8s-prow/tide@sha256:5245b7747c44d560aab27bc07dbaaf50bbb55f71d0973f85b09c79b8d8b93c97",
+		},
+		{
+			name:               "app.ci with public domain",
+			ctx:                "app.ci",
+			potentiallyPrivate: "gcr.io/k8s-prow/tide@sha256:5245b7747c44d560aab27bc07dbaaf50bbb55f71d0973f85b09c79b8d8b93c97",
+			expected:           "gcr.io/k8s-prow/tide@sha256:5245b7747c44d560aab27bc07dbaaf50bbb55f71d0973f85b09c79b8d8b93c97",
+		},
+		{
+			name:               "unknown context",
+			ctx:                "unknown",
+			potentiallyPrivate: "gcr.io/k8s-prow/tide@sha256:5245b7747c44d560aab27bc07dbaaf50bbb55f71d0973f85b09c79b8d8b93c97",
+			expected:           "",
+			expectedError:      fmt.Errorf("failed to get the domain for context unknown and service registry"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual, actualError := publicDomainForImage(tc.ctx, tc.potentiallyPrivate)
+			if diff := cmp.Diff(tc.expected, actual); diff != "" {
+				t.Errorf("actual does not match expected, diff: %s", diff)
+			}
+			if diff := cmp.Diff(tc.expectedError, actualError, equateErrorMessage); diff != "" {
+				t.Errorf("actualError does not match expectedError, diff: %s", diff)
+			}
+		})
+	}
+}
+
+func TestFindNewest(t *testing.T) {
+	now := time.Now()
+	testCases := []struct {
+		name          string
+		isTags        map[string]*imagev1.ImageStreamTag
+		expected      string
+		expectedError error
+	}{
+		{
+			name:          "nil isTags",
+			expectedError: fmt.Errorf("cannot find the newest isTag from map[]"),
+		},
+		{
+			name:          "empty isTags",
+			isTags:        map[string]*imagev1.ImageStreamTag{},
+			expectedError: fmt.Errorf("cannot find the newest isTag from map[]"),
+		},
+		{
+			name: "both nil",
+			isTags: map[string]*imagev1.ImageStreamTag{
+				"ctx1": nil,
+				"ctx2": nil,
+			},
+			expectedError: fmt.Errorf("cannot find the newest isTag from map[ctx1:nil ctx2:nil]"),
```

Comment:
I would just make the underlying func return a `found bool` instead of an error

Reply:
Update.